main.cpp

```diff
@@ -19,15 +19,33 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter): _rewriter(rewriter) {};
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+    void run(const MatchFinder::MatchResult &Result) override 
+    {
+        const auto *expression = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");

```

Comment:
What if `expression` is a null pointer? How to protect the program from crashing in such cases?

Reply:
We check for nullptr:

if (expression == nullptr) {
    return;
}